File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/service/SessionService.java
package com.vipamp.vipclaw.admin.service;

import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import com.baomidou.mybatisplus.extension.service.IService;
import com.vipamp.vipclaw.admin.dto.SessionCreateRequest;
import com.vipamp.vipclaw.admin.dto.SessionResponse;
import com.vipamp.vipclaw.admin.entity.Session;
import org.springframework.lang.Nullable;

/**
 * 会话服务接口
 *
 * @author vipamp
 * @since 2026-03-25
 */
public interface SessionService extends IService<Session> {

    /**
     * 分页查询会话列表
     *
     * @param keyword 模糊查询字段
     * @param status  状态筛选字段
     * @param current 当前页码
     * @param size    每页大小
     * @return 分页结果
     */
    Page<Session> getSessionPage(@Nullable String keyword, @Nullable Integer status, Integer current, Integer size);

    /**
     * 获取单个会话详情
     *
     * @param id 会话 ID
     * @return 会话实体
     */
    Session getSessionById(Long id);

    /**
     * 创建会话
     *
     * @param request 会话创建请求对象
     * @return 创建结果
     */
    boolean createSession(SessionCreateRequest request);

    /**
     * 切换会话启用状态
     *
     * @param id     会话 ID
     * @param status 启用状态（0:禁用，1:启用）
     * @return 更新结果
     */
    boolean toggleSessionStatus(Long id, Integer status);

    /**
     * 删除会话
     *
     * @param id 会话 ID
     * @return 删除结果
     */
    boolean deleteSession(Long id);

    /**
     * 将 Session 实体转换为响应 DTO（包含完整的技能和 MCP 信息）
     *
     * @param session 会话实体
     * @return 响应 DTO
     */
    SessionResponse convertToResponse(Session session);
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/service/SessionService.kt
package com.vipamp.vipclaw.admin.service

import com.vipamp.vipclaw.admin.dto.Page
import com.vipamp.vipclaw.admin.dto.SessionChatUpdateRequest
import com.vipamp.vipclaw.admin.dto.SessionCreateRequest
import com.vipamp.vipclaw.admin.dto.SessionResponse
import com.vipamp.vipclaw.admin.entity.Session
import com.vipamp.vipclaw.ascopagent.dto.SessionConfigResponse

/**
 * Session service interface
 */
interface SessionService {

    /**
     * Query session list with pagination
     *
     * @param keyword  Fuzzy search field
     * @param status   Status filter field
     * @param pageNum  Current page number
     * @param pageSize Page size
     * @return Paginated result
     */
    fun page(keyword: String?, status: Int?, pageNum: Int, pageSize: Int): Page<Session>

    /**
     * Get single session details
     *
     * @param id Session ID
     * @return Session entity
     */
    fun getSession(id: Long): Session?

    /**
     * Create session
     *
     * @param request Session create request object
     * @return Create result
     */
    fun createSession(request: SessionCreateRequest): Boolean

    /**
     * Update session
     *
     * @param id      Session ID
     * @param request Session update request object
     * @return Update result
     */
    fun updateSession(id: Long, request: SessionCreateRequest): Boolean

    /**
     * Update session configuration
     *
     * @param sessionId Session ID
     * @param request   Session config update request object
     */
    fun updateSessionChatConfig(sessionId: String, request: SessionChatUpdateRequest)

    /**
     * Get session configuration
     *
     * @param sessionId Session ID
     * @return Session config response object
     */
    fun getSessionChatConfig(sessionId: String): SessionConfigResponse

    /**
     * Toggle session enable status
     *
     * @param id     Session ID
     * @param status Enable status (0:disabled, 1:enabled)
     * @return Update result
     */
    fun toggleSessionStatus(id: Long, status: Int): Boolean

    /**
     * Delete session
     *
     * @param id Session ID
     * @return Delete result
     */
    fun deleteSession(id: Long): Boolean

    /**
     * Check if session name exists
     *
     * @param title Session name
     * @return Whether exists
     */
    fun existsByTitle(title: String): Boolean

    /**
     * Convert Session entity to response DTO (including complete skill and MCP information)
     *
     * @param session Session entity
     * @return Response DTO
     */
    fun convertToResponse(session: Session): SessionResponse
}
